modules/user_data/variables.tf

```diff
@@ -1,23 +1,51 @@
 
-variable "fqdn" {}
+variable "fqdn" {
+  description = "The fully qualified domain name of the load balancer."
+  type        = string
+}
 variable "active_active" {
-  default = true
-  type    = bool
-}
-variable "generated_bastion_key_private" {}
-variable "aws_bucket_bootstrap" {}
-variable "aws_bucket_data" {}
-variable "aws_region" {}
-variable "tfe_license" {}
-variable "kms_key_arn" {}
+  default     = true
+  description = "A boolean that indicates if the TFE deployment is Active/Active or Standalone."
+  type        = bool
+}
+variable "generated_bastion_key_private" {
+  description = "The PEM formatted private data of the SSH key pair associated with the bastion EC2 instance."
+  type        = string
+}
+variable "aws_bucket_bootstrap" {
+  description = "The name of the S3 storage bucket which contains TFE bootstrap artifacts."
+  type        = string
+}
+variable "aws_bucket_data" {
+  description = "The name of the S3 storage bucket which contains TFE runtime data."
+  type        = string
+}
+variable "aws_region" {
+  description = "The region in which the S3 storage bucket which contains TFE runtime data is deployed."
+  type        = string
+}
+variable "tfe_license" {
+  description = "The name of the S3 storage bucket object which contains the TFE license."
+  type        = string
+}
+variable "kms_key_arn" {
+  description = "The Amazon Resource Name of the KMS key which is used to encrypt S3 storage bucket objects."
+  type        = string
+}
 variable "redis_host" {
-  default = ""
+  default     = ""
+  description = "The IP address of the primary node in the Redis Elasticache replication group."
+  type        = string
 }
 variable "redis_pass" {
-  default = ""
+  default     = ""
+  description = "The password which is required to create connections with the Redis Elasticache replication group."
+  type        = string
 }
 variable "redis_port" {
-  default = ""
+  default     = ""
+  description = "The port number on which the Redis Elasticache replication group accepts connections."
+  type        = string
```

Comment:
Just making sure that the empty string defaults are on purpose.

Reply:
I think we should consider updating them to be `null`, but that may require other changes beyond the scope of this pull request.

Reply:
I don't think this one should be `null` cause it gets concatenated with the host and passed as a string to the config.